pandas/core/reshape/reshape.py

```diff
@@ -205,6 +205,9 @@ def get_new_values(self, values, fill_value=None):
 
         # we can simply reshape if we don't have a mask
         if mask_all and len(values):
+            # TODO: Under what circumstances can we rely on sorted_values
```

Comment:
I think if the we are sorted *already* then you can simplify, but I am not sure we can make that guarantee.